packages/rocket-auth-aws-infrastructure/src/lambdas/sign-in.ts

```diff
@@ -0,0 +1,37 @@
+import { CognitoIdentityServiceProvider } from 'aws-sdk'
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda'
+import { errorResponse, okResponse } from './response'
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  try {
+    const params = JSON.parse(event.body!)
+    const isCustom = process.env.mode === 'Passwordless'
+    const cognitoService = new CognitoIdentityServiceProvider()
+    const initAuthResponse = await cognitoService
+      .initiateAuth({
+        AuthFlow: isCustom ? 'CUSTOM_AUTH' : 'USER_PASSWORD_AUTH',
+        ClientId: process.env.userPoolClientId!,
+        AuthParameters: {
+          USERNAME: params.username,
+          PASSWORD: isCustom ? params.username : params.password,
+        },
+      })
+      .promise()
+
+    if (isCustom) {
+      return okResponse({
+        session: initAuthResponse.Session,
+        message: 'Use the session and the code we have sent you via SMS to get your access tokens via POST /token.',
+      })
+    }
+    return okResponse({
+      accessToken: initAuthResponse.AuthenticationResult?.AccessToken,
+      idToken: initAuthResponse.AuthenticationResult?.IdToken,
+      expiresIn: initAuthResponse.AuthenticationResult?.ExpiresIn,
+      refreshToken: initAuthResponse.AuthenticationResult?.RefreshToken,
+      tokenType: initAuthResponse.AuthenticationResult?.TokenType,
+    })
+  } catch (e) {
+    return errorResponse(e)
+  }
```

Comment:
I see that many of these lambdas are nearly equal, perhaps you could create a wrapper higher order function in order to remove duplication?

Reply:
Good idea @NickSeagull, I'll review all the code again and I will apply some suggestions. I'll let you know.